### PR TITLE
enable community phrase to mitigate spam

### PR DESCRIPTION
**Pull request by DGaffney**

### CODE REVIEW

**Comment by DGaffney** [comment]
Hello @aeschright! @turoczy is having to delete multiple spam comments on the calagator. I'm thinking that this PR may be able to fix it albeit kinda sloppily. Idea here is we add a `community_phrase` that is set as an env var - this phrase is seeded with trusted members of the community, and shared readily to people known to be a part of the community. As we see more spam coming back, we rotate this as need be. Kinda gatekeepy! But also the simplest possible thing to 100% lock out spammers with (hopefully) minimal impact to the good users...

File: app/controllers/calagator/application_controller.rb [hunk 1]
```diff
@@ -92,5 +97,11 @@ def recaptcha_verified?(model)
 
       true
     end
+
+    def community_phrase_verified?(model)
+      return verify_recaptcha(model: model) if recaptcha_enabled?
```
Comment (ryanwi) [inline app/controllers/calagator/application_controller.rb:102]: Did you mean to call the verify_recaptcha method inside here?

Comment (DGaffney) [reply app/controllers/calagator/application_controller.rb:102]: I did not. I'll fix it now.

**Comment by ryanwi** [comment]
Interesting idea. 

Also, I am curious what the state of recaptcha is too for battling spam? Are spammers defeating it?

**Comment by DGaffney** [comment]
> Interesting idea.
> 
> Also, I am curious what the state of recaptcha is too for battling spam? Are spammers defeating it?

The state is that when it's used in it's stock setup, it's defeated by an API request that costs $0.0005 per solution (see https://2captcha.com/ - I use it as part of my stack at [IPM](https://ipm-corporation.com/)). Very likely the spammers are just solving with one of these tools and eating the cost, or doing it in-house for "free".

**Comment by reidab** [comment]
Hi @DGaffney — thanks for putting together this proposal. 

One of our core principles from the very beginning of the project has always been that allowing open edits make for a better experience, as anyone can fix incorrect information, enhance listings, or remove inappropriate content. Especially for the Portland Calagator instance, this is something we've stuck to despite it not always being the simplest answer. 

While I appreciate the sentiment of trying to make this community phrase an "open secret" that's easy for people to gain access to, the idea of having to know the right people to get into the edit club is just gatekeepy enough that I really wouldn't want to implement it on https://calagator.org without exhausting other options first.

That said, spam has always been a problem of varying degree and I think it's worth discussing. I've spun out a discussion here https://github.com/calagator/calagator/discussions/701 with all the history of what we've tried and other ideas I know about. Let's see if we can figure out something with our current tools or a different direction.

I'm going to close this PR now. If we do decide that this is a way to move in the future, we can always come back to it.

**Comment by reidab** [comment]
Hi @DGaffney — thanks for putting together this proposal. 

One of our core principles from the very beginning of the project has always been that allowing open edits make for a better experience, as anyone can fix incorrect information, enhance listings, or remove inappropriate content. Especially for the Portland Calagator instance, this is something we've stuck to despite it not always being the simplest answer. 

While I appreciate the sentiment of trying to make this community phrase an "open secret" that's easy for people to gain access to, the idea of having to know the right people to get into the edit club is just gatekeepy enough that I really wouldn't want to implement it on https://calagator.org without exhausting other options first.

That said, spam has always been a problem of varying degree and I think it's worth discussing. I've spun out a discussion here https://github.com/calagator/calagator/discussions/701 with all the history of what we've tried and other ideas I know about. Let's see if we can figure out something with our current tools or a different direction.

I'm going to close this PR now. If we do decide that this is a way to move in the future, we can always come back to it.

**Pull request closed without merging**